File: 0145/solution.ts
class TreeNode {
  val: number;
  left: TreeNode | null;
  right: TreeNode | null;
  constructor(val?: number, left?: TreeNode | null, right?: TreeNode | null) {
    this.val = val === undefined ? 0 : val;
    this.left = left === undefined ? null : left;
    this.right = right === undefined ? null : right;
  }
}

function postorderTraversal(root: TreeNode | null): number[] {
  const res: number[] = [];
  const stack: TreeNode[] = [];
  if (root) {
    stack.push(root);
  }
  while (stack.length) {
    let curr = stack.pop()!;
    res.push(curr.val);
    if (curr.left) {
      stack.push(curr.left);
    }
    if (curr.right) {
      stack.push(curr.right);
    }
  }
  res.reverse();
  return res;
}

File: 0145/solution.go
package main

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func postorderTraversal(root *TreeNode) []int {
	var res []int
	traverse(root, &res)
	return res
}

func traverse(node *TreeNode, post *[]int) {
	if node == nil {
		return
	}

	traverse(node.Left, post)
	traverse(node.Right, post)
	*post = append(*post, node.Val)
}
